backend/battle/serializers.py

```diff
@@ -0,0 +1,177 @@
+from rest_framework import serializers
+
+from battle.models import Battle, Team, PokemonTeam
+from battle.tasks import run_battle_and_send_result_email
+from battle.battles.battle import save_pokemons_if_they_dont_exist
+
+from services.battle import (
+    has_pokemons_fields_missing_values,
+    has_fields_with_wrong_pokemon_name,
+    has_positions_fields_missing_values,
+    has_pokemon_sum_valid,
+    has_positions_fields_with_duplicate_values,
+    has_different_contenders_to_creator_and_opponent,
+    fetch_opponent_or_create_if_doesnt_exist,
+    send_invite_email_or_send_password_email
+)
+
+from users.models import User
+
+from pokemon.models import Pokemon
+
+POSITION_CHOICES = [(1, 1), (2, 2), (3, 3)]
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ("email", "id")
+
+
+class TeamSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Team
+        fields = ("id", "battle", "trainer", "pokemons")
+
+
+class TeamCreateSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+
+    battle = serializers.PrimaryKeyRelatedField(queryset=Battle.objects.all())
+    trainer = serializers.PrimaryKeyRelatedField(queryset=User.objects.all())
+
+    pokemon_1 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+    pokemon_2 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+    pokemon_3 = serializers.CharField(style={'base_template': 'textarea.html'}, required=False)
+
+    position_pkn_1 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+    position_pkn_2 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+    position_pkn_3 = serializers.ChoiceField(
+        choices=POSITION_CHOICES,
+        label="Position",
+        required=False)
+
+    class Meta:
+        model = Team
+        fields = (
+            "battle",
+            "trainer",
+            "position_pkn_1",
+            "position_pkn_2",
+            "position_pkn_3",
+            "pokemon_1",
+            "pokemon_2",
+            "pokemon_3"
+        )
+
+    def validate(self, attrs):
+
+        if Team.objects.filter(battle=attrs['battle'], trainer=attrs['trainer']):
```

Comment:
A better way is: `Team.objects.filter(battle=attrs['battle'], trainer=attrs['trainer']).exists()`.

Reply:
thanks :D